examples/hello_world_c/BUILD.bazel

```diff
@@ -1,4 +1,33 @@
+load("@aspect_bazel_lib//lib:transitions.bzl", "platform_transition_filegroup")
+load("@bazel_tools//tools/build_defs/pkg:pkg.bzl", "pkg_tar")
+
 cc_binary(
     name = "hello_world_c",
     srcs = ["main.c"],
 )
+
+platform_transition_filegroup(
```

Comment:
nit: the examples are more useful if they have some comments

Reply:
I'm reworking the examples as tests in another branch, so this is only a temporary assertion.